glean-core/android/src/main/java/mozilla/telemetry/glean/Dispatchers.kt

```diff
@@ -142,29 +133,13 @@ internal object Dispatchers {
                     // Task is a suspending function.
                     task()
                 }
-            }
 
-            // In order to ensure that the queued tasks are executed in the proper order, we will
-            // wait up to 5 seconds for it to complete, otherwise we will reset the flag so that
-            // new tasks may continue to run.
-            runBlocking {
-                if (withTimeoutOrNull(QUEUE_PROCESSING_TIMEOUT_MS) {
-                    job.join()
-                } == null) {
-                    Log.e(LOG_TAG, "Timeout processing initial tasks queue")
-                    GleanError.preinitTasksTimeout.set(true)
-                    queueInitialTasks.set(false)
-                    taskQueue.clear()
-                } else {
-                    Log.i(LOG_TAG, "Initial tasks queue completed successfully")
+                // This must happen after `queueInitialTasks.set(false)` is run, or it
+                // would be added to a full task queue and be silently dropped.
+                if (overflowCount > 0) {
+                    GleanError.preinitTasksOverflow.addSync(MAX_QUEUE_SIZE + overflowCount)
```

Comment:
Note: since the tasks are now dispatched asynchronously, this is now a sync call within the "flush" job.